whelk-core/src/main/groovy/whelk/search/XLQLQuery.java

```diff
@@ -424,6 +427,40 @@ public static String quoteIfPhraseOrContainsSpecialSymbol(String s) {
         return s.matches(".*(>=|<=|[=!~<>(): ]).*") ? "\"" + s + "\"" : s;
     }
 
+    public Map<String, Object> pAgg(Entity object, Disambiguate.OutsetType outsetType) {
+        Map<String, Object> query = new LinkedHashMap<>();
+
+        var filters = curatedPredicates(object)
+                .stream()
+                .collect(Collectors.toMap(
+                        Function.identity(),
+                        p -> getEsQuery(new QueryTree(new SimpleQueryTree(pvEqualsLink(p, object.id)), disambiguate, outsetType, esMappings.nestedFields))));
+
+        if (!filters.isEmpty()) {
+            query.put("_p", Map.of("filters", Map.of("filters", filters)));
+        }
+
+        return query;
+    }
+
+    private List<String> curatedPredicates(Entity object) {
+        // TODO don't hardcode
+        Map<String, List<String>> typeToRelations =
+                Map.of(
+                        "Agent", List.of("subject", "contributor", "publisher"),
+                        "Concept", List.of("subject", "genreForm", "hasOccupation", "fieldOfActivity"),
+                        "Work", List.of("subject", "contributor")
```

Comment:
`[] :contributor [ a :Work ]` seems odd! Guess we'll evaluate these declarations when we move them to configurations anyway.

Reply:
Yes, copy-paste typo. Fixed.
Yes.